rl.py

```diff
@@ -182,71 +320,133 @@ def sec_to_hours(seconds):
     print("-------------------------------\n")
     start_time = time.time()
 
-    if TRAIN:
-        while idx < n_epochs[args.num_env]:
-
-           try:
-
-               print("\n-------------------------------")
-               print(">>>    Start training n°", idx+1)
-               print("[INFO]  >>    time: ", sec_to_hours(time.time()-start_time))
-               print("[INFO]  >>    scene: ", id)
-               print("[INFO]  >>    algo: ", algo)
-               print("[INFO]  >>    seed: ", seed)
-               print("-------------------------------\n")
-
-               model.learn(total_timesteps=timer_limit*20, log_interval=20)
-               model.save("./Results_benchmark/" + name + "/latest")
-
-               print("\n-------------------------------")
-               print(">>>    Start test n°", idx+1)
-               print("[INFO]  >>    scene: ", id)
-               print("[INFO]  >>    algo: ", algo)
-               print("[INFO]  >>    seed: ", seed)
-               print("-------------------------------\n")
-
-               r, final_r = test(test_env, model, idx, n_test=5)
-               final_rewards.append(final_r)
-               rewards.append(r)
-               steps.append(timer_limit*20*(idx+1))
-
-
-               with open("./Results_benchmark/" +  name + "/rewards_"+id+".txt", 'w') as fp:
-                   json.dump([rewards, steps], fp)
-               with open("./Results_benchmark/" +  name + "/final_rewards_"+id+".txt", 'w') as fp:
-                   json.dump([final_rewards, steps], fp)
+    if TRAIN == 'new':
```

Comment:
Again polymorphism must be used. At minimum, it must be separated in 2 functions.